Herebyfile.mjs

```diff
@@ -833,11 +832,19 @@ export const produceLKG = task({
         if (missingFiles.length > 0) {
             throw new Error("Cannot replace the LKG unless all built targets are present in directory 'built/local/'. The following files are missing:\n" + missingFiles.join("\n"));
         }
-        const sizeBefore = getDirSize("lib");
+
+        /** @type {number | undefined} */
+        let sizeBefore;
+        if (fs.existsSync("lib")) {
+            sizeBefore = getDirSize("lib");
+        }
         await exec(process.execPath, ["scripts/produceLKG.mjs"]);
-        const sizeAfter = getDirSize("lib");
-        if (sizeAfter > (sizeBefore * 1.10)) {
-            throw new Error("The lib folder increased by 10% or more. This likely indicates a bug.");
+
+        if (sizeBefore !== undefined) {
+            const sizeAfter = getDirSize("lib");
+            if (sizeAfter > (sizeBefore * 1.10)) {
+                throw new Error("The lib folder increased by 10% or more. This likely indicates a bug.");
+            }
         }

```

Comment:
This all is really unfortunate because it won't let us catch regressions in size until release anymore, because on main there won't be a lib folder.

I might need to change the `smoke` test task to build `lib` on main before checking this so we can still get the size checks. Otherwise we won't detect if we accidentally add a bunch of code to the package.

Reply:
Working on fixing this in CI now; forgive the weird commits.

Reply:
Frankly i thought thats what smoke test did

Reply:
Any time anyone runs `LKG`, the size check happens, so it does happen here, yeah, though this test does more; it also validates that the format of the output is expected.

That latter bit should really be a script but I didn't do it that way when I fixed the issue it's testing; should do that too.

Reply:
That being said, this task really shouldn't be "the thing" that does this.

In a followup, I'll remove some of this and instead add a specific CI task we can use that tracks package sizing specifically.

Reply:
~~36699下！他/~~